File: test-2-vul-1.py
import sqlite3
from flask import Flask, request

app = Flask(__name__)

@app.route("/user")
def get_user():
    username = request.args.get("username")

    # 🚨 Vulnerable: unsanitized input in SQL query
    query = f"SELECT * FROM users WHERE username = '{username}'"
    conn = sqlite3.connect('users.db')
    cursor = conn.cursor()
    cursor.execute(query)  # CodeQL will flag this
    result = cursor.fetchall()

    return {"result": result}

File: test-2-vul-1.js
const http = require('http');
const url = require('url');

http.createServer(function (req, res) {
    const queryObject = url.parse(req.url, true).query;
    const userCode = queryObject.code;

    // 🚨 Vulnerability: eval with unsanitized user input
    eval(userCode); // CodeQL will flag this

    res.writeHead(200, {'Content-Type': 'text/plain'});
    res.end('Code executed');
}).listen(8080);
